src/noaa_flares.py

```diff
@@ -127,86 +192,204 @@ def http_ngdc(self):
+            month_string = filename[4:6]
+            day_string = filename[6:8]
+            try:
+                file_date = datetime(year=int(year_string),
+                                     month=int(month_string),
+                                     day=int(day_string))
+            except ValueError:
+                # Probably a goes-xrs-report which is always more than 3 days old
+                #print(f"Can't determine file date for {filename}")
+                continue
+            #print(f"{filename} day: {day_string} month: {month_string} year: {year_string} "
+            #        f"filedate: {file_date}")
+            file_age = current_time - file_date
+            #print(f"{file_age.days} days old")
+            if file_age.days > 4:
+                #print(f"File {filename} is greater than 4 days old")
+                continue
+
+            # Don't delete file if downloaded less than an hour ago
+            modification_timestamp = os.path.getmtime(file_path)
+            modification_time = datetime.fromtimestamp(modification_timestamp)
+            current_time = datetime.now()
+            file_age = current_time - modification_time
+            #print(f"{file_path} is {file_age.seconds} seconds old")
+            if file_age.seconds < 3600: # seconds is 1 hour
+                #print(f"{file_path} is less than 1 hour  old")
+                continue
+
+            try:
+                os.remove(file_path)
+                print(f"Deleted: {file_path}")
+            except OSError as err:
+                print(f"Error deleting {file_path}: {err}")
+        print("Done purging cache")
 
 # Run some test cases
 if __name__ == '__main__':
 
     test_list = ["20140104",
-                 "20170104",
-                 "20201211",
-                 "20250529",
-                 datetime(2023, 10, 1, 12, 0),
-                 datetime(2023, 10, 1, 12, 0, tzinfo=timezone.utc),
-                 date(2023, 10, 1),
-                 "2023123a"]  # this one should throw an exception
-
-
-    print("NOAA_flares test cases")
-    print("eventName, BeginTime, MaxTime, EndTime, Particulars")
+                  "20130525", # this one should download the goes-xrs report
+                  "20130525", # should be in the cache so faster
+                  "20170104",
+                  "20201211",
+                  "20250529",
+                 # # next 3 have the same date and file so 1st
+                 # # should download, 2nd and 3rd use cache
+                  datetime(2023, 10, 1, 12, 0),
+                  datetime(2023, 10, 1, 12, 0, tzinfo=timezone.utc),
+                  date(2023, 10, 1),
+
+                 datetime.now(timezone.utc),
+                 "foobar"]  # invalid input - throw exception
+
+    def clear_xra_cache():
+        """ Utility to clear the XRA cache before running tests"""
+        directory_path = script_relative_to_cwd_relative(path.join("..", "Private"))
```

Comment:
Shouldn't the cache path be handled in analogy `to __init__()`. Here is a copy of the referenced code.
This would also need an additional function parameter `noaa_cache_path=None`
```
        if noaa_cache_path is None:
            self.cache_path = script_relative_to_cwd_relative(path.join("..", "Private"))
        else:
            self.cache_path = noaa_cache_path
```